### ISSUE TITLE
in function findOrCreate   Cannot convert undefined or null to object

### ISSUE THREAD

**Issue by amwybj041989** [sequelize/sequelize#11216]
## What are you doing?
When the aim is to create an account, first find create again

```js
async regist() {
    const ctx = this.ctx;
    let params = ctx.request.body;
    if (!params.hasOwnProperty("account")) {
      return (ctx.response.body = statusCode.ERROR_400("account"));
    }
    if (!params.hasOwnProperty("password")) {
      return (ctx.response.body = statusCode.ERROR_400("password"));
    }
    await ctx.model.User.findOrCreate({
      where: {
        account: params.account
      },
      defaults: {
        account: params.account,
        password: MD5(params.password)
      },
    }).then(([data, res]) => {
      // if (!res) {
      //   return (ctx.response.body = statusCode.ERROR_202());
      // }
      // return (ctx.response.body = statusCode.SUCCESS());
    });
  }
```
**To Reproduce**
1. used postman requested, 
2. if res==false,not error
3. if The database data already exists,error:

nodejs.TypeError: Cannot convert undefined or null to object
--

_Output, either JSON or SQL_

## Environment
Dialect:
- [x ] mysql
- [ ] postgres
- [ ] sqlite
- [ ] mssql
- [ ] any
Dialect **library** version: XXX
Database version: 5.2
Sequelize version: 5.2.0
Node Version: 10.16.0

**Comment by papb** [comment]
Please provide a [MVCE](https://stackoverflow.com/help/minimal-reproducible-example). We need full code that can be copy-pasted and executed to see the error.

**Comment by Gahen** [comment]
Maybe this will help someone: in my case I found this was an error when trying to report the actual sequelize error, an that error was an attempt to insert over an unique key. 

I think it may be because 2 requests to findOrCreate are done almost at the same time, so the 2nd one tries to create an entity that the first one just created.

Here's where node reported the error occurred
```javascript
      }).catch(sequelizeErrors.UniqueConstraintError, err => {
        const flattenedWhere = Utils.flattenObjectDeep(options.where);
        const flattenedWhereKeys = Object.keys(flattenedWhere).map(name => _.last(name.split('.')));
        const whereFields = flattenedWhereKeys.map(name => _.get(this.rawAttributes, `${name}.field`, name));
        const defaultFields = options.defaults && Object.keys(options.defaults)
          .filter(name => this.rawAttributes[name])
          .map(name => this.rawAttributes[name].field || name);

        const errFieldKeys = Object.keys(err.fields);  // HERE was the error reported
        const errFieldsWhereIntersects = Utils.intersects(errFieldKeys, whereFields);
```

**Comment by Lennac** [comment]
> Maybe this will help someone: in my case I found this was an error when trying to report the actual sequelize error, an that error was an attempt to insert over an unique key.
> 
> I think it may be because 2 requests to findOrCreate are done almost at the same time, so the 2nd one tries to create an entity that the first one just created.
> 
> Here's where node reported the error occurred
> 
> ```js
>       }).catch(sequelizeErrors.UniqueConstraintError, err => {
>         const flattenedWhere = Utils.flattenObjectDeep(options.where);
>         const flattenedWhereKeys = Object.keys(flattenedWhere).map(name => _.last(name.split('.')));
>         const whereFields = flattenedWhereKeys.map(name => _.get(this.rawAttributes, `${name}.field`, name));
>         const defaultFields = options.defaults && Object.keys(options.defaults)
>           .filter(name => this.rawAttributes[name])
>           .map(name => this.rawAttributes[name].field || name);
> 
>         const errFieldKeys = Object.keys(err.fields);  // HERE was the error reported
>         const errFieldsWhereIntersects = Utils.intersects(errFieldKeys, whereFields);
> ```

I'm experiencing this exact same error, but not sure how a second request for findOrCreate is being made in my code. I can supply code snippets if required.

**Comment by ortonomy** [comment]
This issue has come up once before as far as I can tell, here: https://github.com/sequelize/sequelize/issues/9003

There seems to be a huge bug in Sequelize where if you define your own column names for IDs in your models, and then use `findOrCreate` on an empty table, it *will not work*

**Comment by ortonomy** [comment]
It's definitely not stale and still a problem!

**Comment by ortonomy** [comment]
It's not stale!

**Comment by WikiRik** [comment]
Yeah, the bot is quite aggressive. I have added the `type: bug`  label and now it won't mark it as stale anymore. It would be great if you could provide a [MVCE](https://stackoverflow.com/help/minimal-reproducible-example) or make a PR with a test that fails so it's easier to find a fix for it

**Comment by GabrielFerrariR** [comment]
The problem persists on sequelize v6
